core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1364,6 +1385,16 @@ private[spark] class DAGScheduler(
                   if (job.numFinished == job.numPartitions) {
                     markStageAsFinished(resultStage)
                     cleanupStateForJobAndIndependentStages(job)
+                    try { // cancelTasks will fail if a SchedulerBackend does not implement killTask
+                      logInfo(
+                        s"Job ${job.jobId} is finished. Killing speculative tasks for this job")
```

Comment:
message should be updated as this should be more then speculative tasks as it could be tasks in other attempts.